### PR TITLE
Google auth

### PR DESCRIPTION
**Pull request by EugenGam**

### CODE REVIEW

File: routers/google.controller.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        const token = await foundUser.generateAndSaveToken();
+        callback(null, { token });
+      }
+    } catch (err) {
+      callback(err, null);
+    }
+  }
+
+  redirectGoogle(req, res) {
+    res.status(201).send({ token: req.user.token }).redirect('/');
```
Comment (alucardpetrovich) [inline routers/google.controller.js]: Token in query parameter of redirect (<redirect_url>?token=<token_value>). Also redirect link is link to front-end. Your variant won't work.

File: routers/google.controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const { userModel } = require('../database/models');
+
+class GoogleController {
+  constructor() {}
+  async findOrCreate(profile, callback) {
+    try {
+      const { email } = profile._json;
+      const userName = email.substring(0, email.indexOf('@'));
```
Comment (alucardpetrovich) [inline routers/google.controller.js]: `displayName`, I guess, would be better

File: routers/google.controller.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      const foundUser = await userModel.findOne({ email });
```
Comment (alucardpetrovich) [inline routers/google.controller.js]: We could do findOrCreate in one mongoDB request with `findOneAndUpdate + upsert + $setOnInsert`

File: helpers/google-cred.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+const configEnv = require('../config.env');
+
+module.exports = {
+  clientID: configEnv.google.clientId,
+  clientSecret: configEnv.google.clientSecret,
+  callbackURL: `http://localhost:${configEnv.port}/auth/google/callback`,
```
Comment (ihahalev) [inline helpers/google-cred.js]: as I understand it should be ${configEnv.srvUrl}

File: routers/google.controller.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+const { userModel } = require('../database/models');
+
+class GoogleController {
+  constructor() {}
+  async findOrCreate(profile, callback) {
+    try {
+      const { email } = profile._json;
+      const displayName = email.substring(0, email.indexOf('@'));
+      const user = userModel.update(
+        email,
```
Comment (alucardpetrovich) [inline routers/google.controller.js]: I'm pretty sure that .update does not return user. Also I've told about `findOneAndUpdate`

**Review by ihahalev — APPROVED** [review]

**Pull request merged**